### DISCUSSION TITLE
SMA Home Manager 2.0

### DISCUSSION THREAD

**Discussion by Snoopy-HSS** [hoylabs/OpenDTU-OnBattery#508] [Ideas]
Hi,

habe SMA HM 2.0 an die OpenDTU angeflanscht.
Interesse die library hier zu integrieren?

Grüße

HSS

**Comment by TheMikeCAD** [comment]
Servus,
auf jeden Fall!

**Comment by Snoopy-HSS** [comment]
Sage bescheid wenn du was brauchts

**Comment by TheMikeCAD** [comment]
Servus,
super nett von Dir!
Also die Überlegung ist, das System ohne "Balkon-PV" AC zu koppeln.
Könnte man dann doch einfach als AC Speicher an eine große PV hängen?! Wenn Du dann auch noch die Einbindung des SMA SHM hättest, könne man da auch auf einen Raspberry verzichten.. wie genial ist das denn?!
PN geht hier leider nicht - oder?

**Comment by TheMikeCAD** [comment]
könntest Du bitte die Einbindung bereitstellen?

**Comment by helgeerbe** [comment]
Dazu gibt es auch einen PR #518 von @Snoopy-HSS den ich kommentiert habe. So kann ich ihn leider nicht einfach mergen. Ich habe ihn deswegen auch erstmal auf draft gesetzt.

**Reply by Snoopy-HSS** [reply]
Sag mir was du brauchst...

**Reply by Snoopy-HSS** [reply]
Habe einen neuen PULL Request eingestellt auf Basis der aktuellen Source

**Comment by TheMikeCAD** [comment]
ok Freunde, ich bewundere Euch dafür, was ihr hier leistet / könnt.. aber da bin ich leider raus

**Comment by Elektron79** [comment]
Hi @Snoopy-HSS,

so wie ich die Implementierung aktuell verstehe, kann Deiner Erweiterung die Mulitcast-Pakete (UDP) vom SunnyPortal-HM2.0 empfangen und damit die Leistung der Wechselrichter per openDTU so einstellen, dass eine Nulleinspeisung erfolgt.
Hast Du auch mal über den umgekehrten Weg nachgedacht? Also die Wechselrichter an der openDTU wie einen „großen“ SMA Wechselrichter (ggf.  It Leistungsbegrenzung) aussehen zu lassen?

Dazu gibt es hier auf Github schon Beispiele wie das geht, alleine mir fehlen die Fähigkeiten als Entwickler um das zu integrieren. 

Würde mir konkret ersparen einen SMA-EM zu kaufen und alle (auch die SMA Wechselrichter) über dieses Smartmeter zu messen…

viele Grüße Michael

**Comment by Snoopy-HSS** [comment]
Hi Michael.

Viele Stunden probiert.

Als Wechselrichter geht nicht, lässt SMA nicht zu.

Als SMA-EM funktioniert. Dazu habe ich Bibliotheken. Du kannst allerdings nur einen EM im SMA Portal haben. Wechselrichter + EM geht nicht, not supported…

Mehrere EM geht nicht, alle Erzeuger müssen immer hinter einem EM sein, egal ob SMA oder nicht…

Grüße

HSS

Am 31.12.2023 um 18:31 schrieb Elektron79 ***@***.***>:

﻿

Hi @Snoopy-HSS<https://github.com/Snoopy-HSS>,

so wie ich die Implementierung aktuell verstehe, kann Deiner Erweiterung die Mulitcast-Pakete (UDP) vom SunnyPortal-HM2.0 empfangen und damit die Leistung der Wechselrichter per openDTU so einstellen, dass eine Nulleinspeisung erfolgt.
Hast Du auch mal über den umgekehrten Weg nachgedacht? Also die Wechselrichter an der openDTU wie einen „großen“ SMA Wechselrichter (ggf. It Leistungsbegrenzung) aussehen zu lassen?

Dazu gibt es hier auf Github schon Beispiele wie das geht, alleine mir fehlen die Fähigkeiten als Entwickler um das zu integrieren.

Würde mir konkret ersparen einen SMA-EM zu kaufen und alle (auch die SMA Wechselrichter) über dieses Smartmeter zu messen…

viele Grüße Michael

**Reply by Elektron79** [reply]
Hallo HSS,

schade, aber dann bliebe ja die Möglichkeit ein EM zu emulierten und die Daten von den (bestehenden) SMA Wechselrichtern zum Ertrag der Hoymiles Wechselrichter zu addieren.
Ich habe aktuell im Bestand SMA Wexhselrichter die direkt mit dem SHM2.0 kommunizieren.
Wenn ich nun mit den Hymiles erweitere muß ich alle Wechselrichter über ein zusätzliches EM2.0 führen der dann der Erzeugungszähler wird.

Mit dieser Erweiterung könnte dann das zusätzliche EM2.0 entfallen. Die Konfiguration wäre identisch, aber der ESP wäre dann das EM. Die Werte der bestehenden SMA-Wexhselrichter kommen ja entweder der Modbus oder per Multicast.

viele Grüße Michael

**Comment by BytePhilHD** [comment]
@Snoopy-HSS @helgeerbe Wäre es möglich dass ihr das irgendwie integriert? 
Also dass der OpenDTU als SMA Energy Meter erkannt wird?

**Reply by Snoopy-HSS** [reply]
Wie Elektron79 schon beschrieben hat.
Wenn ein EM2 im System ist, müssen alle Wechselrichter dahinter sein.

Ich würde das per MQTT lösen und dort den EM2 emulieren.
Ansonsten müsste ja die OPENdtu noch die SMA Wechselrichter auslesen um deren Werte zu erhalten.
